rosplan_planning_system/src/ProblemGeneration/ProblemGeneratorFactory.cpp

```diff
@@ -10,6 +10,7 @@ namespace KCL_rosplan {
     ProblemGeneratorPtr ProblemGeneratorFactory::createProblemGenerator(ProblemGeneratorFactory::ProbGen pg_type, const std::string& kb) {
         if (pg_type == ProbGen::PDDL) return ProblemGeneratorPtr(new PDDLProblemGenerator(kb));
         else if (pg_type == ProbGen::RDDL) return ProblemGeneratorPtr(new RDDLProblemGenerator(kb));
+        else if (pg_type == ProbGen::CHIMP) return ProblemGeneratorPtr(new CHIMPProblemGenerator(kb));
```

Comment:
Why is CHIMP here but HDDL in the KB? 
Same with the problem generator. Does CHIMP have its own problem format which is different from HDDL?

Reply:
Yes, exactly. CHIMP has its own format for problems and domains.

Reply:
And is this format HDDL then? My main question was the difference in names for the KB and the PG, I think this should be consistent (if the KB and the PG use the same kind of files).

Reply:
No, CHIMP extensions are ".ddl" for the domain and ".pdl" for the problem. However in future we want to support HDDL planners, so for now we are defining 2 domains, 1 in HDDL, 1 in CHIMP format, KB loads the domain in HDDL format and the chimp problem generator reads from KB and generates a problem in CHIMP format. In future HDDL problem generators are expected, and in general full support for HDDL planners.

Reply:
Gerard mentioned: it would be nice to have an alias CHIMP KB, which is just the HDDL knowledge base. This would be clearer overall. We can put this on a todo list?